src/test/java/uk/gov/pay/directdebit/webhook/gocardless/services/GoCardlessEventHandlerForMandatesTest.java

```diff
@@ -48,43 +56,52 @@
             .withPayerFixture(payerFixture);
 
     private GoCardlessEventFixture goCardlessEventFixture = GoCardlessEventFixture.aGoCardlessEventFixture()
+            .withResourceType(GoCardlessResourceType.MANDATES)
             .withLinksOrganisation(organisationIdentifier);
 
     @Test
     public void handle_onAFailedMandateGoCardlessEvent_shouldRegisterAPayEventAsMandateFailed() {
-        GoCardlessEvent goCardlessEvent = spy(goCardlessEventFixture.withAction("failed").toEntity());
+        GoCardlessEvent goCardlessEvent = goCardlessEventFixture.withAction("failed").toEntity();
 
         when(mockMandateQueryService.findByGoCardlessMandateIdAndOrganisationId(
                 goCardlessEvent.getLinksMandate().get(), goCardlessEventFixture.getLinksOrganisation()))
                 .thenReturn(mandateFixture.toEntity());
 
-        goCardlessMandateHandler.handle(goCardlessEvent);
+        goCardlessEventHandler.handle(List.of(goCardlessEvent));
 
         verify(userNotificationService).sendMandateFailedEmailFor(mandateFixture.toEntity());
+        verify(mockMandateQueryService).findByGoCardlessMandateIdAndOrganisationId(
+                goCardlessEvent.getLinksMandate().get(), goCardlessEventFixture.getLinksOrganisation());
```

Comment:
We shouldn’t need to verify this method call because it’s already stubbed. If it isn’t called, the test should fail anyway.

Reply:
fixed